service/ciphering/aesEncoder.go

```diff
@@ -51,17 +53,20 @@ func (en aesEncoder) Encrypt(plaintext string, aesKey []byte) (encryptedText str
 		return "", err
 
 	}
-	nonce := make([]byte, gcm.NonceSize())
 
-	if _, err = io.ReadFull(rand.Reader, nonce); err != nil {
+	nonce, err := en.np.generate(gcm.NonceSize())
+
+	if err != nil {
 		return "", err
 	}
-	encryptedText = hex.EncodeToString(gcm.Seal(nonce, nonce, []byte(plaintext), nil))
+	encryptedText = hex.EncodeToString(gcm.Seal([]byte(nonce), []byte(nonce), []byte(plaintext), nil))
 	return encryptedText, nil
 
 }
 
-func NewAESEncoder() *aesEncoder {
+func NewAESEncoder(np nonceProducer) *aesEncoder {
```

Comment:
With experiments and analysis. But you can start with a question do you still need `regularEncoder` implementation? Could you suggest changes that would allow using `aesEncoder` with the same purpose as `regularEncoder`?